lms/djangoapps/instructor/views/api.py

```diff
@@ -3362,3 +3369,102 @@ def _get_boolean_param(request, param_name):
+
+    openassessment_blocks = modulestore().get_items(course_key, qualifiers={'category': 'openassessment'})
+    if len(openassessment_blocks):
+        responses = OraAggregateData.collect_ora2_responses(course_id)
+        for block in openassessment_blocks:
+            if block.parent not in parents:
+                parents[block.parent] = modulestore().get_item(block.parent)
+            result_item_id = unicode(block.location)
+            staff_assessment = "staff-assessment" in block.assessment_steps
+            result_item = {
+                'id': result_item_id,
+                'name': block.display_name,
+                'parent_id': unicode(block.parent),
+                'parent_name': parents[block.parent].display_name,
+                'total': 0,
+                'staff_assessment': staff_assessment,
+                'url': reverse('open_response_assessment_block_display', kwargs={
+                    'course_id': unicode(course_key),
+                    'ora_block_id': result_item_id
+                })
+            }
+            total = 0
+            if result_item_id in responses:
+                result_item.update(responses[result_item_id])
+            for step in ora2_steps:
+                if step in result_item:
+                    total += result_item[step]
+                else:
+                    result_item[step] = 0
+            result_item['total'] = total
+            results.append(result_item)
+
+    return JsonResponse(results)
+
+
+@transaction.non_atomic_requests
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def open_response_assessment_block_display(request, course_id, ora_block_id):
+    """
+    Returns the rendered view of a given ORA XBlock, with related resources.
+
+    :param request: HttpRequest object
+    :param course_id: course identifier
+    :param ora_block_id: chosen ORA block identifier
+    :return: JsonResponse object
+    """
+    is_staff_view = request.GET.get('is_staff', False)
+    ora_block_id = unquote(ora_block_id)
+    usage_key = UsageKey.from_string(ora_block_id)
+    course_key = CourseKey.from_string(course_id)
+    view_name = 'student_view' if not is_staff_view else 'grade_available_responses_view'
+
+    with modulestore().bulk_operations(course_key):
+        # verify the user has access to the course, including enrollment check
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+
+        # get the block, which verifies whether the user has access to the block.
+        block, _ = get_module_by_usage_id(
+            request, unicode(course_key), unicode(usage_key), disable_staff_debug_info=True, course=course
+        )
+
+        if block.category != 'openassessment':
+            return HttpResponseForbidden()
+
+        try:
+            fragment = block.render(view_name, context=request.GET)
```

Comment:
Wow, that's at least the third time this logic has been implemented! There was an open source PR to create a centralized way to do this but it never landed:

https://github.com/edx/edx-platform/pull/11605#issuecomment-224666123

I've been thinking that we should move such a function into the web-fragments repo so that any project can do this. We could then consolidate all the places that do this (the ones I know are Studio, the above PR, and https://github.com/edx-solutions/jquery-xblock). It is on my giant list of things I'd like to do, but you know how that goes.